_posts/2021-10-09-software-and-projects-as-systems.md

```diff
@@ -0,0 +1,191 @@
+
+### What are the motivation and goals of each user role?
+
+A key take-away here is users want to spend as little time as possible with the technical system.
+
+Software developers should forget their own interests and peer groups if those like to fiddle with things. That's not normal :)
+
+Software developers may be distracted from this cornerstone because the time of a developer is spent building all those shiny features. Users have their real work to do and there the specific software application is only one smallish step towards their business goal. Users do not want to spend their time learning every this 'n that about a software application.
+
+To recap: changes to the system may have unexpected results if the business goal of each application user role is not clear.
+
+### What are the motivation and goals of IT specialists?
+
+What level is their knowledge about the service operated?
+
+What level is their knowledge about the business domain of the service operated?
+
+IT specialists usually do not work full time with a specific product or service. They have a bunch of applications and services which to keep up and running somehow. Each very different and serving different business domains. Often domains which they are only superficially familiar with.
+
+Changes to the system may have unexpected results if the IT operations are not considered. For example, does a change impact following IT operations:
+
+- How application is configured and what is the likelihood of misconfiguration
+- Logging procedures and analysis
+- Version updating procedures
+- Backup procedures
+- Rollback procedures
+- System health monitoring
+
+## The Producing System
+
+![Charlie Chaplin CC BY-SA 2.0](/img/software-and-project-as-systems/charlie-chaplin-modern-times.jpg "Charlie Chapling in a mahine in Modern times")
+([Image source](#image-sources))
+
+Systems produced by software developers, or their tool systems, are not the only systems they are working with. Software developers themselves are part of:
+
+- a project or a service team or several
+- a higher-level organization or several
+
+### Project
+
+Project is an endeavor with the following attributes - also known as the project iron triangle:
+
+- Schedule
+- Budget
+- Goal
+
+An old rule is that not all three should be fixed, but that is a subject for a separate discussion.
+
+Project itself produces deliveries to meet the project goals. Deliveries are groups of specified and approved changes which affect people called stakeholders.
+
+Changes to the system may have unexpected results if all those are not considered:
+
+- A project has a very high risk of failing or producing waste if schedule, goals and budget are not clear and agreed by project stakeholders.
+- Evaluation and approval of deliveries is difficult if goals are not clear.
+- For example
+  - Software developers should prioritize work differently when all mandatory goals are already met than if delivery is only halfway ready.
```

Comment:
- Software developers should prioritize work differently when all mandatory goals are already met than if delivery is only halfway ready.

Than on tässä mielestäni väärä sana näiden vertailuun, mutta en nyt osaa myöskään ehdottaa suoraan korjausta.